# GA0012

## Title
Invalid usage of OnlyPermitSpecifiedTypesAttribute

## Category
Validity

## Severity
Error

## Details
This error is emitted when a type parameter's explicit type constraints do not permit the usage of any type while including `OnlyPermitSpecifiedTypesAttribute`.

## Example
```csharp
class C
<
    [ProhibitedTypes(typeof(int), typeof(long))]
    // GA0012 will appear here
    [OnlyPermitSpecifiedTypes]
    T
>
{ }
```
