# GA0040

## Title
Invalid type group permission clashing with exclusive type group permission

## Category
Validity

## Severity
Error

## Details
The type groups that are exclusively permitted will never allow other type groups to be permitted.

See how the type group filtering system works at the [docs](../usage/type-group-filters.md).

## Example
```csharp
class GenericType
<
    [FilterArrayTypes(FilterType.Exclusive)]
    // GA0040 will appear here
    [FilterGenericTypes(FilterType.Permitted)]
    // GA0040 will appear here
    [FilterAbstractClasses(FilterType.Permitted)]
    T
>;
```