# GA0019

## Title
Invalid type parameter name

## Category
Validity

## Severity
Error

## Group
[DG03](groups/DG03.md)

## Details
This error is emitted when the referred type parameter name does not exist in the declaring member's type parameter list.

## Example
```csharp
class C
<
    // GA0019 will appear here
    [InheritTypeConstraints("T1")]
    T
>
{ }
```
