# GA0052

## Title
Invalid type group specialization

## Category
Validity

## Severity
Error

## Details
The specialization value is not valid for this type group.

See how the type group filtering system works at the [docs](../usage/type-group-filters.md).

## Example
```csharp
class GenericType
<
    // GA0052 will appear here
    [FilterGenericTypes(FilterType.Prohibited, 0)]
    // GA0052 will appear here
    [FilterArrayTypes(FilterType.Prohibited, 0)]
    [FilterGenericTypes(FilterType.Permitted)]
    [FilterArrayTypes(FilterType.Permitted)]
    T
>;
```